packages/@glimmerx/eslint-plugin-glimmerx/lib/rules/template-vars.js

```diff
@@ -0,0 +1,81 @@
+const { traverse, preprocess } = require('@glimmer/syntax');
+const { getTemplateTokens } = require('@glimmerx/babel-plugin-component-templates');
+
+module.exports = {
+  docs: {
+    description: 'Components / Helpers referenced in hbs template literals should not trigger no-unused-vars failures, but should trigger no-undef if they are not defined propery',
+    category: 'Variables',
+    recommended: false, // Currently this rule is not on by default due to the Single File Component usage not being the default
+  },
+  meta: {
+    messages: {
+      undefToken: 'Token {{ token }} is used in an hbs tagged template literal, but is not defined'
+    },
+    // example: '@glimmerx/glimmerx/template-vars': [2, 'unused-only', { nativeTokens: ['anImplicitToken'] }]
+    schema: [
+      // Configures whether to only prevent no-unused-vars errors ('unused-only), or also throw an eslint error when a template token is undefined ('all')
+      {
+        enum: ['unused-only', 'all'],
+        default: 'all'
+      },
+      {
+          type: 'object',
+          properties: {
+            // ['if', 'each'] will always be 'nativeTokens', but you may add more via this configuration. One use-case is if a token is added to the
+            // Javascript code implicitly (such as via a babel transform)
+            nativeTokens: {
+              type: 'array',
+              items: {
+                type: 'string'
+              }
+            }
+          },
+          additionalProperties: false
+      }
+    ],
+  },
+  create(context) {
+    const defaultNativeTokens = ['if', 'each'];
+    let isGlimmerSfc = false;
+    return {
+      'ImportDeclaration'(node) {
+        if (isGlimmerSfc || node.source.value !== '@glimmerx/component') {
+          return;
+        }
+        node.specifiers.forEach((specifier) => {
+          const identifier = specifier.name || specifier.local;
+          if (identifier.name === 'hbs') {
+            isGlimmerSfc = true;
+          }
```

Comment:
this needs to be able to handle renames of the import so for example: `import {hbs as _hbs} from '@glimmerx/component'`